common.py

```diff
@@ -1,27 +1,26 @@
 import jinja2
 from dataclasses import dataclass, fields, field, is_dataclass
 from enum import Enum
-from typing import List, Optional, Any, Dict, List, Union, Type, TypeVar, cast
+from typing import List, Optional, Any, Dict, List, Union, Type, TypeVar, Generic, cast
 
 FT_BASE_IMG = "quay.io/wizhao/ft-base-image:0.9"
 TFT_TOOLS_IMG = "quay.io/wizhao/tft-tools:latest"
 TFT_TESTS = "tft-tests"
 
 
-def enum_factory(enum_type: Type[Enum]):
-    def factory(value: Any) -> Enum:
-        if isinstance(value, enum_type):
-            return value
-        elif isinstance(value, str):
-            member = enum_type.__members__.get(value)
-            if member is not None:
-                return member
-            else:
-                raise ValueError(f"{value} is not a valid member of {enum_type}")
-        else:
-            raise ValueError(f"Cannot convert {value} to {enum_type}")
+E = TypeVar("E", bound=Enum)
+
 
-    return factory
+def enum_convert(enum_type: Type[E], value: Any) -> E:
```

Comment:
ah, I commented about the `enum_factory` already. Maybe it would be good to rebase the branch to not first introduce code that gets modified/fixed by a followup value.

Still, I think it should be `value: str | E`.